usr/lib/cca_stdll/cca_specific.c

```diff
@@ -7357,6 +7357,344 @@ static CK_RV cca_handle_apqn_event(STDLL_TokData_t *tokdata,
     return CKR_OK;
 }
 
+struct cca_affected_data {
+    struct hsm_mk_change_info *info;
+    CK_BBOOL affected;
+};
+
+static CK_RV cca_mk_change_is_affected_cb(STDLL_TokData_t *tokdata,
+                                          const char *adapter,
+                                          unsigned short card,
+                                          unsigned short domain,
+                                          void *private)
+{
+    struct cca_affected_data *ad = private;
+
+    UNUSED(tokdata);
+
+    if (hsm_mk_change_apqns_find(ad->info->apqns, ad->info->num_apqns,
+                                 card, domain)) {
+        TRACE_DEVEL("%s APQN %02X.%04X (%s) is affected by MK change\n",
+                    __func__, card, domain, adapter);
+        ad->affected = TRUE;
+    }
+
+    return CKR_OK;
+}
+
+static CK_RV cca_mk_change_is_affected(STDLL_TokData_t *tokdata,
+                                       struct hsm_mk_change_info *info)
+{
+    unsigned int i;
+    CK_BBOOL affected = FALSE;
+    struct cca_affected_data ad;
+    CK_RV rc;
+
+    for (i = 0; i < info->num_mkvps; i++) {
+        TRACE_DEVEL("%s MK type: %d\n", __func__, info->mkvps[i].type);
+        switch (info->mkvps[i].type) {
+        case HSM_MK_TYPE_CCA_SYM:
```

Comment:
Why is HSM_MK_TYPE_CCA_ASYM not considered here? Also below.

Reply:
The ASYM master key is a 3DES key used to encrypt older RSA key token versions. 3DES is considered weak, so such older RSA key token versions should not be used anymore. Newer RSA key tokens are encrypted with the APKA master key which is an AES-256 key. 
Opencryptoki does not generate any such older RSA key tokens, so the ASYM master key is not used by OpenCryptoki anymore, and thus we don't need to consider the ASYM master key here. 
Note that the pkcshasm_mk_change tool DOES support the ASYM master key, because it is token independent, and it does not know if there is (or will be) a token that uses the ASYM master key.